### PR TITLE
test: Add tests for VertexAIChatGeminiGenerator and migrate from preview

### LINKED ISSUE #1018: Migration from vertex_ai.preview to vertex_ai.generative_model
## Summary and motivation

Current integration of VertexAIGemini uses `vertex_ai.preview` package which has some experimental features. We should migrate to a more stable `vertetx_ai.generative_model` package. 

## Checklist

```[tasklist]
### Tasks
- [ ] The changes are merged in the `main` branch (Code + Docstrings)
- [ ] New package version declares the breaking change
- [ ] The package has been released on PyPI
- [ ] Docs at https://docs.haystack.deepset.ai/ were updated
- [ ] Integration tile on https://github.com/deepset-ai/haystack-integrations was updated
- [ ] Notebooks on https://github.com/deepset-ai/haystack-cookbook were updated (if needed)
- [ ] Tutorials on https://github.com/deepset-ai/haystack-tutorials were updated (if needed)
- [ ] Articles on https://github.com/deepset-ai/haystack-home/tree/main/content were updated (if needed)
```

### PR DESCRIPTION
**Pull request by Amnah199**
### Related Issues

- fixes: #1019 
- fixes #1018 

### Proposed Changes:

- Migrate from `preview` package to stable version
- Add a new test file for `VertexAIChatGeminiGenerator`
- Add pipeline serialization/deserialization test

### How did you test it?

- Ran all the tests

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CODE_OF_CONDUCT.md)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.

### CODE REVIEW

**Comment by Amnah199** [comment]
@julian-risch worst case scenario, merging changes with testing failing only on windows is an option?

**Comment by davidsbatista** [comment]
You should also add a serialization/deserialization test of a pipeline containing a VertexAIChatGeminiGenerator

**Comment by davidsbatista** [comment]
@Amnah199 here's an example of the type of test I mentioned above:

https://github.com/deepset-ai/haystack-experimental/blob/main/test/components/retrievers/test_auto_merging_retriever.py#L103

File: integrations/google_vertex/tests/chat/test_gemini.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+                ],
+                "streaming_callback": None,
+            },
+        }
+    )
+
+    assert gemini._model_name == "gemini-1.5-flash"
+    assert gemini._project_id == "TestID123"
+    assert gemini._safety_settings == {HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT: HarmBlockThreshold.BLOCK_ONLY_HIGH}
+    # assert repr(gemini._tools) == repr([Tool(function_declarations=[GET_CURRENT_WEATHER_FUNC])])
```
Comment (vblagoje) [inline integrations/google_vertex/tests/chat/test_gemini.py:233]: Why comment out?

**Review by vblagoje — COMMENTED** [review]
Left one comment about tool definition

File: integrations/google_vertex/tests/chat/test_gemini.py (hunk 1 above)
Comment (Amnah199) [reply integrations/google_vertex/tests/chat/test_gemini.py]: Was by mistake. Just fixed it 👍

**Review by vblagoje — APPROVED** [review]
LGTM

**Pull request merged**